NeuroID/src/debug/java/com/neuroid/tracker/storage/NIDDataStoreManager.kt

```diff
@@ -260,4 +173,91 @@ private object NIDDataStoreManagerImp : NIDDataStoreManager {
 
         return jsonBody.toString()
     }
+
+    /**
+     * Convert a list of NIDEventModel to a list of JSONObject.  The provided list will be cleared
+     * during the conversion.  Where necessary, events are updated to fill in missing values.
+     */
+    private fun toJsonList(list: MutableList<NIDEventModel>): List<JSONObject> {
+        val jsonList = mutableListOf<JSONObject>()
+        while (list.isNotEmpty()) {
+            val event = list.removeAt(0)
+            event.accel?.let {
+                it.x = it.x ?: NIDSensorHelper.valuesAccel.axisX
+                it.y = it.y ?: NIDSensorHelper.valuesAccel.axisY
+                it.z = it.z ?:NIDSensorHelper.valuesAccel.axisZ
+            }
+            event.gyro?.let {
+                it.x = it.x ?: NIDSensorHelper.valuesGyro.axisX
+                it.y = it.y ?: NIDSensorHelper.valuesGyro.axisY
+                it.z = it.z ?:NIDSensorHelper.valuesGyro.axisZ
+            }
+            if (event.type == "CREATE_SESSION" && event.url == "") {
+                event.url = "$ANDROID_URI${NIDServiceTracker.firstScreenName}"
+            }
+            jsonList.add(event.getJSONObject())
+        }
+        return jsonList
+    }
+
+    @Synchronized
+    override fun addViewIdExclude(id: String) {
+        if (listIdsExcluded.none { it == id }) {
+            listIdsExcluded.add(id)
+        }
+    }
+
+    /**
+     * Debug logging for captured events.
+     */
+    private fun logEvent(event: NIDEventModel) {
+        NeuroID.getInstance()?.captureIntegrationHealthEvent(event = event)
+
+        if (!BuildConfig.DEBUG) {
+            return
+        }
```

Comment:
We should still log even on release packages. The customer determines this with the `enableLogging` command

Reply:
In 3.0.4, all log methods are preceded by a `BuildConfig.DEBUG` check.